samples/boards/nrf/mesh/onoff_level_lighting_vnd_app/src/main.c

```diff
@@ -94,13 +94,8 @@ static void light_default_status_init(void)
 void update_vnd_led_gpio(void)
 {
 #ifndef ONE_LED_ONE_BUTTON_BOARD
-	if (vnd_user_data.current == STATE_ON) {
-		/* LED2 On */
-		gpio_pin_write(led_device[1], DT_ALIAS_LED1_GPIOS_PIN, 0);
-	} else {
-		/* LED2 Off */
-		gpio_pin_write(led_device[1], DT_ALIAS_LED1_GPIOS_PIN, 1);
-	}
+	gpio_pin_set(led_device[1], DT_ALIAS_LED1_GPIOS_PIN,
+		     vnd_user_data.current == STATE_ON);
```

Comment:
We could write it as `vnd_user_data.current == STATE_ON ? 1 : 0` to be Misra-C compliant. A few more `gpio_pin_set` calls may need to be updated. Non blocking.